ui/src/components/MacroForm.tsx

```diff
@@ -0,0 +1,362 @@
+                  onMoveDown={() => handleStepMove(stepIndex, 'down')}
+                  onKeySelect={(option) => handleKeySelect(stepIndex, option)}
+                  onKeyQueryChange={(query) => handleKeyQueryChange(stepIndex, query)}
+                  keyQuery={keyQueries[stepIndex] || ''}
+                  onModifierChange={(modifiers) => handleModifierChange(stepIndex, modifiers)}
+                  onDelayChange={(delay) => handleDelayChange(stepIndex, delay)}
+                  isLastStep={stepIndex === (macro.steps?.length || 0) - 1}
+                />
+              ))}
+            </div>
+          </Fieldset>
+
+          <div className="mt-4">
+            <Button
+              size="MD"
+              theme="light"
+              fullWidth
+              LeadingIcon={LuPlus}
+              text={`Add Step ${isMaxStepsReached ? `(${MAX_STEPS_PER_MACRO} max)` : ''}`}
+              onClick={() => {
+                if (isMaxStepsReached) {
+                  showTemporaryError(`You can only add a maximum of ${MAX_STEPS_PER_MACRO} steps per macro.`);
+                  return;
+                }
+                
+                setMacro(prev => ({
+                  ...prev,
+                  steps: [
+                    ...(prev.steps || []), 
+                    { keys: [], modifiers: [], delay: DEFAULT_DELAY }
+                  ],
+                }));
+                setErrors({});
+              }}
+              disabled={isMaxStepsReached}
+            />
+          </div>
+
+          {errorMessage && (
+            <div className="mt-4">
+              <FieldError error={errorMessage} />
+            </div>
+          )}
+
+          <div className="mt-6 flex items-center justify-between">
+            {showCancelConfirm ? (
+              <div className="flex items-center gap-2">
+                <span className="text-sm text-slate-600 dark:text-slate-400">
+                  Cancel changes?
+                </span>
+                <Button
+                  size="SM"
+                  theme="danger"
+                  text="Yes"
+                  onClick={onCancelConfirm}
+                />
+                <Button
+                  size="SM"
+                  theme="light"
+                  text="No"
+                  onClick={() => onCancel()}
+                />
+              </div>
+            ) : (
+              <>
+                <div className="flex gap-x-2">
+                  <Button
+                    size="SM"
+                    theme="primary"
+                    text={isSubmitting ? "Saving..." : submitText}
+                    onClick={handleSubmit}
+                    disabled={isSubmitting}
+                  />
+                  <Button
+                    size="SM"
+                    theme="light"
+                    text="Cancel"
+                    onClick={onCancel}
+                  />
+                </div>
+                {showDelete && (
```

Comment:
I noticed, was reaching for this button when I wanted to Save the Macro. I would suggest adding it to the other side of the `SettingsPageHeader` in `devices.$id.settings.macros.add.tsx`. 

Can you also add the `LuTrash2` to it, I really like that icon.

Reply:
yeah cool, removed the delete button from the macro form, added it to the edit route top right